src/views/forms/FormTransferTransaction/FormTransferTransactionTs.ts

```diff
@@ -278,6 +273,27 @@ export class FormTransferTransactionTs extends FormTransactionBase {
         }
         this.formItems.recipient = !!this.recipient ? this.recipient : null;
 
+        this.formItems.messagePlain = this.message ? Formatters.hexToUtf8(this.message.payload) : '';
+        this.formItems.encryptMessage = false;
+        this.encyptedMessage = null;
+        // - maxFee must be absolute
+        this.formItems.maxFee = this.defaultFee;
+
+        // transaction details passed via router
+        this.importTransaction = this.$route.params.transaction || this.importedTransaction ? true : false;
+        this.resetMosaics(this.importTransaction);
+        this.triggerChange();
+    }
+
+    /**
+     * Resetting mosaics list when signer is changed
+     */
+    private resetMosaics(importedTransaction: boolean): void {
```

Comment:
Should `resetMosaics` call `triggerChanges` conditionally? In the case of a selected mosaic change after the reset, then the fee calculation might change, not 100% sure though.

If so maybe having a second parameter `triggerChanges = false(default)` should be added; and at the end of the method, we can have the following.

```
if (triggerChanges) {
    this.triggerChange();
}
```

Reply:
In that case, Do we need the condition and an extra param? we could move the` triggerChange()` call inside `resetMosaics()` and remove it from `resetForm` instead.

Reply:
Yes, that also works, in that case I'd rename the method to `resetMosaicsAndTriggerChange`.